### ISSUE TITLE
Excluding Tasks

### ISSUE THREAD

**Issue by pjmavcom** [mugiwara85/CodeblockCustomizer#55]
So I use the Tasks plugin ([here](https://github.com/obsidian-tasks-group/obsidian-tasks)) and with this plugin enabled, the task list is populated slowly. With this disabled, it is instant. I tried excluding the language by adding 'tasks' to the language list, but that does not help. Neither does adding 'exclude' to the code block, though I am not sure I am defining it correctly.

This is a code block I use for the task list:
```
```tasks
path includes /Schedule
done
filename includes Project1
```
There are only 16 tasks in the list at the moment. It takes about a second for this list, but the master list that holds all the tasks takes 30 seconds to load. Both lists are instant when this is disabled though.
Is there something I am missing to get this to work correctly? Let me know if you need more info. Thanks for the help!

**Comment by mugiwara85** [comment]
Whyt do you mean by "you are not sure if you are defining it correctly?"

If you excluded tasks, either by defining it in the settings, like this:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/65603765/7a1a8134-03d1-4715-bd3e-7e9022c41c19)
or defined the `exclude` parameter in the first line of the code block, then it should look like this way. (Excluded code block, and a normal code block)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/65603765/5d42f5b2-c057-45ad-a4a7-2bfe4a8a31e5)

I don't know this plugin, so you'll have to explain to me, how I can reproduce the issue. Also does this problem occur in editing mode or reading mode, or both?

**Comment by pjmavcom** [comment]
I'm not sure I'm defining it correctly because there are no examples of defining 'exclude' anywhere. And what you posted does not show it either. I assume the first line needs to look like this: ```tasks exclude
Is that what defining it means? If so, there is no change in what happens. 
I did add it to the 'Exclude languages' section, but it did not change anything, it is still slow.
I didn't think to check on reading or editing view, but here is what I found:
In editing mode, the list of tasks populates one at a time, and takes about one second to populate a list of 16 tasks.
In reading mode, it populates all at once, but it takes more than a second for the whole list to show (there is about a one second delay in the list showing).
In both modes, with CodeblockCustomizer disabled, the lists populate almost instantly. It's like for every task it adds, it runs the block through whatever code your plugin has. So for the 16 tasks in that list, it runs it 16 times. That's just what I am thinking. It looks as though, even though it is excluded, it is still processing it.

**Comment by pjmavcom** [comment]
So, I tried a different code block style plugin and I get the same issue. So I don't think there is a bug with this plugin.

**Comment by mugiwara85** [comment]
I am happy to help, but I still don't know this plugin. You'll have to send me some tasks, or the code you are testing with, or explain how to set it up, so I can reproduce the issue.

**Comment by mugiwara85** [comment]
Fixed in version 1.2.8 (sorry it took so long)
